client/src/AppBundle/Command/DocumentSyncCommand.php

```diff
@@ -48,13 +57,15 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                 return;
             }
 
-            $documents = $this->getQueuedDocuments();
+            /** @var QueuedDocumentData[] $documents */
+            $documents = $this->getQueuedDocumentsData();
+
             $output->writeln(count($documents) . ' documents to upload');
 
-            foreach ($documents as $document) {
+            foreach ($documents as &$document) {
```

Comment:
I don't think pass by reference is needed as it's an object.

Reply:
I was getting some out of memory errors when looping over the array of `QueuedDocumentData` so I've tried to minimise anything which I thought could be impacting this. From what I understand, if you don't use a reference in a foreach loop then a copy is made of the variable which over a large loop would start to add up.

Admittedly I also reduced the number of items in the array at the same time so it could just be this that helped but is there a downside to using reference here over a copy?

Reply:
I think a copy is made by default for arrays and scalar types, but not for objects - as far as I know objects are passed by reference by default. That said, the reference here is harmless, just at worst unnecessary.

Reply:
OK I wasn't aware of that - I'll drop the reference if it's not doing anything 👍

Reply:
Hopefully the tests will reveal any obvious failures